geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -74,12 +74,26 @@ public RedisSet(int expectedSize) {
    */
   public RedisSet() {}
 
+  public static Set<byte[]> sunion(RegionProvider regionProvider, List<RedisKey> keys) {
+    MemberSet result = calculateUnion(regionProvider, keys, true);
+    return getSetOpResult(result);
+  }
+
+  private static MemberSet calculateUnion(RegionProvider regionProvider, List<RedisKey> keys,
+      boolean updateStats) {
+    MemberSet union = new MemberSet();
+    for (RedisKey key : keys) {
+      RedisSet curSet = regionProvider.getTypedRedisData(REDIS_SET, key, updateStats);
+      if (curSet.scard() != 0) {
+        union.addAll(curSet.members);
+      }
```

Comment:
I think it should be fine to not check the cardinality of the current set here, since if it's empty, calling `addAll()` with it won't actually result in any work being done.